### ISSUE TITLE
Recall@1 achieves 100% on the Ubuntu dataset

### ISSUE THREAD

**Issue by JuruoMP** [taesunwhang/UMS-ResSel#1]
Hi

I follow the instruction in the README, to be specified
1. download pre-trained and post-trained checkpoints and dataset by running the two scripts
2. run fine-tuning through ``python3 main.py --model bert_post --task_name ubuntu --data_dir data/ubuntu_corpus_v1 --bert_pretrained bert-post-uncased --bert_checkpoint_path bert-post-uncased-pytorch_model.pth --task_type response_selection --gpu_ids "0,1,2,3" --root_dir./ ''
and then the recall@1 on Ubuntu dataset achieves 100% at the second epoch

Could you please tell me is there anything wrong with my operations?

**Comment by taesunwhang** [comment]
Hi
Looks like you are training the BERT+ model (without UMS) on Ubuntu Corpus V1.
I recommend you to check the `log.info` file if there was a sudden increase in response selection loss during training.

**Comment by JuruoMP** [comment]
Yes, I'm running it without UMS
I checked the output from the screen, which shows the response selection loss being 0.000000 since a very early stage.

**Comment by taesunwhang** [comment]
I have checked the code and found observed that the issue occurred in `config/hparams.py` (`cpu_workers` was set to 0).
Sorry for the issue and thanks for pointing it out! Now, the code has been fixed, train the model with the fixed code. 
Please let me know if you have any questions or concerns regarding this matter.

**Comment by JuruoMP** [comment]
Thanks a lot. It seems the bug is fixed.
I'm not clear why #cpu_workers can influence the performance. I didn't see any usage of cpu workers expect the dataloader. Could you please tell me why this hyper param matters?

**Comment by taesunwhang** [comment]
I'm not sure why `cpu_workers` influence performance. It seems that the model is trained properly at the first epoch, but the loss is being zero since the second epoch. Probably, the dataloader may not return the proper batch...
